src/block_if.c

```diff
@@ -122,7 +122,7 @@ static struct blockif_sig_elem *blockif_bse_head;
 #pragma clang diagnostic pop
 
 static ssize_t
-preadv(int fd, const struct iovec *iov, int iovcnt, off_t offset)
+_preadv(int fd, const struct iovec *iov, int iovcnt, off_t offset)
```

Comment:
When building with Xcode 12.2 Beta 4 it fails with the following error:

```
/Users/shane/Code/xhyve/src/block_if.c:248:15: error: 'preadv' is only available on macOS 11.0 or newer [-Werror,-Wunguarded-availability-new]
                        if ((len = preadv(bc->bc_fd, br->br_iov, br->br_iovcnt,
                                   ^~~~~~
In module 'Darwin' imported from /Users/shane/Code/xhyve/src/block_if.c:30:
/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.0.sdk/usr/include/sys/uio.h:103:9: note: 'preadv' has been marked as being introduced in macOS 11.0 here, but the deployment target is macOS 10.10.0
ssize_t preadv(int, const struct iovec *, int, off_t) __DARWIN_NOCANCEL(preadv) __API_AVAILABLE(macos(11.0), ios(14.0), watchos(7.0), tvos(14.0));
        ^
/Users/shane/Code/xhyve/src/block_if.c:248:15: note: enclose 'preadv' in a __builtin_available check to silence this warning
                        if ((len = preadv(bc->bc_fd, br->br_iov, br->br_iovcnt,
                                   ^~~~~~
```

It seems that `preadv` has been added in Big Sur. I couldn't find an easy way of using the builtin function in Big Sur and this shim for earlier versions so I just avoided the name collision. Suggestions for improvement welcome!

Similarly for `pwritev` below.

Reply:
This will be fine for now, as it does not regress any existing behavior. However, it would be nice to make use of `preadv`. I think we can fix this later by checking `__MAC_OS_X_VERSION_MAX_ALLOWED`, like @jeremyhu suggested in #192.

Reply:
I have tried to wrap this function as suggested but I get the same error, it seems like it would want the target SDK to be 11.0+ 🤷🏻‍♂️